### PR TITLE
[37463] Pass in work package project to IUM

### PR DESCRIPTION
**Pull request by oliverguenther**
When showing a work package from a subproject, the invite user modal should be opened for that work package's project, not the current project. Otherwise the user will not be assignable afterwards.

https://community.openproject.org/work_packages/37463

### CODE REVIEW

File: frontend/src/app/modules/autocompleter/create-autocompleter/create-autocompleter.component.ts [hunk 1]
```diff
@@ -104,8 +106,11 @@ export class CreateAutocompleterComponent extends UntilDestroyedMixin implements
     this.onAfterViewInit.emit(this);
     if (this.opInviteUserModalService) {
       this.opInviteUserModalService.close
-        .pipe(this.untilDestroyed())
-        .subscribe((user: HalResource) => {
+        .pipe(
+          this.untilDestroyed(),
+          filter(user => !!user)
```
Comment (oliverguenther) [inline frontend/src/app/modules/autocompleter/create-autocompleter/create-autocompleter.component.ts:111]: This is an unrelated fix I noticed. When cancelling the modal, the assignee field was still saved and a notification was shown.

**Review by b12f — COMMENTED** [review]
Isn't this also used in `multi-select-edit-field` ? I was looking at this earlier today and thought I saw three distinct usages of the button

**Comment by oliverguenther** [comment]
> Isn't this also used in `multi-select-edit-field` ? I was looking at this earlier today and thought I saw three distinct usages of the button

You're right. I added it to the multi select field. For the new dynamic form, we'd need a way to access the project reference of the resource we're passing in. Right now, as you can only edit in the current project, it does not need fixing there. I added a tentative check for a resource's "project" link (if any) so that would work for any resource with an explicit dependency to a project.

File: frontend/src/app/modules/invite-user-modal/button/invite-user-button.component.ts [hunk 2]
```diff
@@ -1,15 +1,20 @@
...
+import { CurrentProjectService } from "core-components/projects/current-project.service";
+import { CurrentUserService } from "core-app/modules/current-user/current-user.service";
 
 @Component({
   selector: 'op-invite-user-button',
   templateUrl: './invite-user-button.component.html',
   styleUrls: ['./invite-user-button.component.sass']
 })
 export class InviteUserButtonComponent implements OnInit {
+  @Input() projectId:string|null;
```
Comment (b12f) [inline frontend/src/app/modules/invite-user-modal/button/invite-user-button.component.ts:16]: Technically this can also be `undefined` since you're using optional chaining

**Review by b12f — APPROVED** [review]
lgtm
